### ISSUE TITLE
[BUG] endless rendering

### ISSUE THREAD

**Issue by eikeon** [petyosi/react-virtuoso#698]
**Describe the bug**

When upgrading to 2.14.0 or 2.16.0 from 2.13.2 the VirtuosoGrid endlessly re-renders calling itemContent repeatedly. 

Perhaps related to the new gap bits?

**Reproduction**
Use https://codesandbox.io/ to illustrate the problem so that I can observe the issue on my side and make sure that a potential fix reliably addresses it.

**To Reproduce**

This is using a VirtuosoGrid as follows:

            <VirtuosoGrid
                useWindowScroll
                overscan={400}
                components={{
                    List: styled.div<GridListProps>`
                        display: grid;
                        gap: 1rem;
                        grid-template-columns: repeat(
                            auto-fill,
                            minmax(var(--gallery-width), 1fr)
                        );
                    `,
                    ScrollSeekPlaceholder: props.scrollSeekPlaceholder,
                }}
                totalCount={props.items.length}
                itemContent={props.itemContent}
                endReached={props.endReached}
                scrollSeekConfiguration={{
                    enter: (velocity, { startIndex, endIndex }) => {
                        setVisibleRange([startIndex, endIndex]);
                        return Math.abs(velocity) > 600;
                    },
                    exit: (velocity) => Math.abs(velocity) < 30,
                    change: (_velocity, { startIndex, endIndex }) =>
                        setVisibleRange([startIndex, endIndex]),
                }}
            />

**Expected behavior**

Expect no re-render when there's no scrolling of the page as was the case in version 2.13.2.

**Desktop (please complete the following information):**

This is on MacOS in both Chrome and Safari

**Comment by petyosi** [comment]
Can you follow the template of the issue and provide a repro in codesandbox?

**Comment by eikeon** [comment]
Yes, I can try and do that when I have a few more minutes. Was hoping this would be helpful in the meantime since it relates to the recent changes. Perhaps along the lines of https://github.com/petyosi/react-virtuoso/pull/692#issuecomment-1166243942

**Comment by eikeon** [comment]
@petyosi Is there an existing codesandbox for VirtuosoGrid that I could key off of?  

BTW, thank you for react-virtuoso ❤️

**Comment by petyosi** [comment]
All code examples from the site have a button which you can use to launch into code sandbox - https://virtuoso.dev/grid-responsive-columns/ - hover the grid to see the little cube at the bottom.

**Comment by eikeon** [comment]
@petyosi Thank you for the quick turnaround! Looking forward to jumping back on `latest` tomorrow

**Comment by eikeon** [comment]
@petyosi And it's running great again 🎉

**Comment by XSPGMike** [comment]
I also had this issue, this fixed it.
